beard/clustering/wrappers.py

```diff
@@ -151,8 +170,26 @@ def fit(self, X, y=None):
                                       criterion=self.criterion,
                                       depth=self.depth, R=self.R,
                                       monocrit=self.monocrit)
-
-                score = self.scoring(y[train], labels[train])
+                if ground_truth:
+                    train = (y != -1)
+
+                    if not self.affinity_score:
+                        score = self.scoring(y[train], labels[train])
+                    else:
+                        # For scoring function that accepts affinity
+                        score = self.scoring(y[train], labels[train], X)
+
+                elif self.affinity_score:
+                    n_labels = len(np.unique(labels))
+                    n_samples = Xs.shape[0]
+
+                    if 1 < n_labels < n_samples:
+                        score = self.scoring(Xs, labels)
+                    else:
+                        score = -np.inf
+                else:
+                    # could have a scoring function that accepts only labels
+                    return self
```

Comment:
Actually, I had that before but I replace it with quitting and added a commit above since I got a test error having the default scoring function "b3_f_score()"  taking exactly 2 arguments. If we don't set a default scoring we will also have this error. The test cases I reach this error with is when we specify the number of clusters or the  treshold. I would need to change the logic of handeling these 2 cases in order to make this change. Is it fine to do that?